### PR TITLE
Show Glyphs Normals

### PR DESCRIPTION
**Pull request by RubioJr9**
This is a debugging tool for glyph normals. To show normals, I had to refactor the glyph constructor class. I extracted the private data members to a struct such that points, lines, and triangles had their data isolated. 

I added 2 new state variables to ShowFieldGlyphs:
ShowNormals(bool)
ShowNormalsScale(double)
Since this is a debugging tool, I didn't make a button for it. You have to set through the python api.

### CODE REVIEW

File: src/Graphics/Glyphs/GlyphConstructor.cc [hunk 1]
```diff
@@ -36,235 +36,270 @@ using namespace Graphics::Datatypes;
...
 {
-  offset_ = numVBOElements_;
-  return offset_;
+  auto& data = getData(prim);
+  data.offset_ = data.numVBOElements_;
+  return data.offset_;
 }
 
-bool GlyphConstructor::normalsValid() const
+bool GlyphConstructor::normalsValid(SpireIBO::PRIMITIVE prim)
```
Comment (dcwhite) [inline src/Graphics/Glyphs/GlyphConstructor.cc:216]:
```suggestion
bool GlyphConstructor::normalsValid(SpireIBO::PRIMITIVE prim) const
```

File: src/Graphics/Glyphs/GlyphConstructor.cc [hunk 2]
```diff
@@ -36,235 +36,270 @@ using namespace Graphics::Datatypes;
...
 {
-  return normals_.size() == points_.size();
+  auto& data = getData(prim);
```
Comment (dcwhite) [inline src/Graphics/Glyphs/GlyphConstructor.cc:218]:
```suggestion
 const auto& data = getData(prim);
```

File: src/Graphics/Glyphs/GlyphConstructor.cc [hunk 3]
```diff
@@ -36,235 +36,270 @@ using namespace Graphics::Datatypes;
...
-  addIndexToOffset(i1);
-  addIndexToOffset(i2);
-  addIndexToOffset(i3);
+  addIndexToOffset(prim, i1);
+  addIndexToOffset(prim, i2);
+  addIndexToOffset(prim, i3);
 }
 
-size_t GlyphConstructor::getCurrentIndex() const
+size_t GlyphConstructor::getCurrentIndex(SpireIBO::PRIMITIVE prim)
```
Comment (dcwhite) [inline src/Graphics/Glyphs/GlyphConstructor.cc:294]:
```suggestion
size_t GlyphConstructor::getCurrentIndex(SpireIBO::PRIMITIVE prim) const
```

File: src/Graphics/Glyphs/GlyphConstructor.cc [hunk 4]
```diff
@@ -36,235 +36,270 @@ using namespace Graphics::Datatypes;
...
 {
-  return numVBOElements_;
+  auto& data = getData(prim);
```
Comment (dcwhite) [inline src/Graphics/Glyphs/GlyphConstructor.cc:296]:
```suggestion
 const auto& data = getData(prim);
```

File: src/Graphics/Glyphs/GlyphConstructor.h [hunk 5]
```diff
@@ -39,43 +39,50 @@
 
 namespace SCIRun {
 namespace Graphics {
+struct GlyphData
```
Comment (dcwhite) [inline src/Graphics/Glyphs/GlyphConstructor.h:42]:
```suggestion
struct SCISHARE GlyphData
```

**Review by dcwhite — CHANGES_REQUESTED** [review]
first wave of line-level review

**Review by jessdtate — APPROVED** [review]

**Review by dcwhite — APPROVED** [review]

**Pull request merged**